cmd/ci-secret-generator/main.go

```diff
@@ -0,0 +1,221 @@
+	Attribute  fieldGenerator `json:"attribute,omitempty"`
+}
+
+type fieldGenerator struct {
+	Name string `json:"name,omitempty"`
+	Cmd  string `json:"cmd,omitempty"`
+}
+
+const (
+	attributeTypePassword string = "password"
+)
+
+func parseOptions() options {
+	var o options
+	flag.CommandLine.BoolVar(&o.dryRun, "dry-run", false, "Whether to actually create the secrets with oc command")
+	flag.CommandLine.StringVar(&o.configPath, "config", "", "Path to the config file to use for this tool.")
+	flag.CommandLine.StringVar(&o.bwUser, "bw-user", "", "Username to access BitWarden.")
+	flag.CommandLine.StringVar(&o.bwPasswordPath, "bw-password-path", "", "Path to a password file to access BitWarden.")
+	flag.CommandLine.StringVar(&o.logLevel, "log-level", "info", fmt.Sprintf("Log level is one of %v.", logrus.AllLevels))
+	flag.CommandLine.IntVar(&o.maxConcurrency, "concurrency", 1, "Maximum number of concurrent in-flight goroutines to BitWarden.")
+	if err := flag.CommandLine.Parse(os.Args[1:]); err != nil {
+		logrus.WithError(err).Errorf("cannot parse args: %q", os.Args[1:])
+	}
+	return o
+}
+
+func (o *options) validateOptions() error {
+	level, err := logrus.ParseLevel(o.logLevel)
+	if err != nil {
+		return fmt.Errorf("invalid log level specified: %w", err)
+	}
+	logrus.SetLevel(level)
+	if o.bwUser == "" {
+		return fmt.Errorf("--bw-user is empty")
+	}
+	if o.bwPasswordPath == "" {
+		return fmt.Errorf("--bw-password-path is empty")
+	}
+	if o.configPath == "" {
+		return fmt.Errorf("--config is empty")
+	}
+	return nil
+}
+
+func (o *options) completeOptions(secrets sets.String) error {
+	bytes, err := ioutil.ReadFile(o.bwPasswordPath)
+	if err != nil {
+		return err
+	}
+	o.bwPassword = strings.TrimSpace(string(bytes))
+	secrets.Insert(o.bwPassword)
+
+	bytes, err = ioutil.ReadFile(o.configPath)
+	if err != nil {
+		return err
+	}
+
+	err = yaml.Unmarshal(bytes, &o.config)
+	if err != nil {
+		return err
+	}
+	return o.validateCompletedOptions()
+}
+
+func (o *options) validateCompletedOptions() error {
+	if o.bwPassword == "" {
+		return fmt.Errorf("--bw-password-file was empty")
+	}
+
+	for i, bwItem := range o.config {
+		if bwItem.ItemName == "" {
+			return fmt.Errorf("config[%d].itemName: empty key is not allowed", i)
+		}
+		switch bwItem.Attribute.Name {
+		case attributeTypePassword, "":
+		default:
+			return fmt.Errorf("config[%d].attribute: only the '%s' is supported, not %s", i, attributeTypePassword, bwItem.Attribute.Name)
+		}
+		if (bwItem.Field.Name != "" && bwItem.Field.Cmd == "") ||
+			(bwItem.Attribute.Name != "" && bwItem.Attribute.Cmd == "") ||
+			(bwItem.Attachment.Name != "" && bwItem.Attachment.Cmd == "") {
+			return fmt.Errorf("config[%d]: empty field not allowed for cmd if name is specified for any of attribute, field or attachment", i)
+		}
+	}
+	return nil
+}
+
+func executeCommand(command string) ([]byte, error) {
+	cmd := strings.Fields(command)
+	out, err := exec.Command(cmd[0], cmd[1:]...).CombinedOutput()
```

Comment:
I think this will panic if `cmd` has a `len` of 1

Reply:
I dont think so - https://play.golang.org/p/XVkmgjMrGEj
lmk if i missed something/this was not what you meant
It would panic for length 0 though - but I think that would be caught in the command empty or not check